tests/unit/APNSClientTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare( strict_types = 1 );
+
+class APNSClientTest extends APNSTest {
+
+	public function testThatRequestsFromNetworkAreProcessed() {
+		$network_service_mock = Mockery::mock( 'MultiplexedNetworkService' );
+		$network_service_mock
+			->shouldReceive( 'enqueueRequest' )
+			->once();
+		$fake_responses = [
+			new Response( 200, $this->new_apns_http_failure_response( 200 ), 1, 1 ),
+			new Response( 400, $this->new_apns_http_failure_response( 400 ), 1, 1 ),
+		];
+		$network_service_mock
+			->shouldReceive( 'sendQueuedRequests' )
+			->andReturn( $fake_responses );
+		$client = new APNSClient( $this->new_configuration(), $network_service_mock );
+
+		$responses = $client->sendRequests( [ $this->new_request() ] );
+
+		$this->assertEquals( 2, count( $responses ) );
```

Comment:
Just to make this a little more robust against future changes, you could consider:

```suggestion
		$this->assertEquals( count($fake_responses), count( $responses ) );
```

But that's really nit-picky of me 😄